logic/broadcaster.js

```diff
@@ -136,34 +122,30 @@ class Broadcaster {
 	 * @todo Add description for the params
 	 */
 	broadcast(params, options, cb) {
-		params.limit = params.limit || this.config.peerLimit;
-		params.broadhash = params.broadhash || null;
+		params.limit = params.limit || this.config.broadcastLimit;
 
 		async.waterfall(
 			[
 				function getPeers(waterCb) {
 					if (!params.peers) {
 						return self.getPeers(params, waterCb);
 					}
-					return setImmediate(waterCb, null, params.peers);
+					return setImmediate(
+						waterCb,
+						null,
+						params.peers.slice(0, params.limit)
+					);
 				},
 				function sendToPeer(peers, waterCb) {
 					library.logger.debug('Begin broadcast', options);
-
-					if (params.limit === self.config.peerLimit) {
-						peers = peers.slice(0, self.config.broadcastLimit);
-					}
-
-					peers
-						.slice(0, self.config.peerLimit)
-						.map(peer => peer.rpc[options.api](options.data));
+					peers.forEach(peer => peer.rpc[options.api](options.data));
 					library.logger.debug('End broadcast');
 					return setImmediate(waterCb, null, peers);
 				},
 			],
 			(err, peers) => {
 				if (cb) {
-					return setImmediate(cb, err, { body: null, peer: peers });
```

Comment:
`body` wasn't used anywhere?

Reply:
No it wasn't. There was a comment earlier to remove it.